src/vmm/src/device_manager/mmio.rs

```diff
@@ -68,7 +68,7 @@ impl fmt::Display for Error {
             }
             Error::EventFd(err) => write!(f, "failed to create or clone event descriptor: {}", err),
             Error::IncorrectDeviceType => write!(f, "incorrect device type"),
-            Error::InternalDeviceError(err) => write!(f, "device error: {}", err),
+            Error::InternalDeviceError(err) => write!(f, "{}", err),
```

Comment:
It is generally better to inline format! arguments (https://rust-lang.github.io/rust-clippy/master/#uninlined_format_args)
```suggestion
            Error::InternalDeviceError(err) => write!(f, "{err}"),
```